### PR TITLE
Fix slow test_patterngen_seeds on Windows by sampling

### LINKED ISSUE #1455: [BUG]: Extremely slow `cuda_core/tests/test_helpers.py::test_patterngen_seeds` on Windows
### Is this a duplicate?

- [x] I confirmed there appear to be no [duplicate issues](https://github.com/NVIDIA/cuda-python/issues) for this bug and that I agree to the [Code of Conduct](CODE_OF_CONDUCT.md)

### Type of Bug

Performance

### Component

cuda.core

### Describe the bug

Using the log archive from https://github.com/NVIDIA/cuda-python/actions/runs/20871705999 (main @ acc78f7c0e570b59fed7b07429dce5bfe143ffd8 ← #1368)

Maximum measured times per platform:
* `linux-64`: 14.93s
* `linux-aarch64`: 12.07s
* `win-64`: **945.29s** (15m 45s)

The `linux-64` maximum (14.93s) is from `Test_linux-64___py3.13__13.0.2__local__rtxpro6000.txt`.

The linux-aarch64 maximum (12.07s) is from `Test_linux-aarch64___py3.14t__13.1.0__local__a100.txt`.

The win-64 maximum (945.29s) is from `Test_win-64___py3.14__13.1.0__local__l4__MCDM_.txt`.

```
smc120-0004.ipp2a2.colossus.nvidia.com:/wrk/main_logs_20871705999 $ grep ' call.*test_helpers.py::test_patterngen_seeds' *.txt
Test_linux-64___py3.10__12.9.1__local__v100.txt:2026-01-10T03:19:17.6039085Z 14.63s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-64___py3.10__13.0.2__wheels__l4.txt:2026-01-10T03:08:24.5435099Z 12.83s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-64___py3.10__13.1.0__wheels__l4.txt:2026-01-10T03:08:21.9057669Z 13.24s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-64___py3.11__12.9.1__wheels__rtxpro6000.txt:2026-01-10T03:19:08.2634608Z 13.43s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-64___py3.11__13.0.2__local__l4.txt:2026-01-10T03:12:47.9856996Z 12.80s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-64___py3.11__13.1.0__local__l4.txt:2026-01-10T03:09:34.1176450Z 12.16s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-64___py3.12__13.0.2__wheels__l4.txt:2026-01-10T03:08:35.6711874Z 12.89s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-64___py3.12__13.1.0__wheels__l4.txt:2026-01-10T03:08:41.5389490Z 13.49s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-64___py3.13__12.9.1__wheels__v100.txt:2026-01-10T03:08:35.8080212Z 13.84s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-64___py3.13__13.0.2__local__H100.txt:2026-01-10T03:10:39.8398063Z 12.63s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-64___py3.13__13.0.2__local__rtxpro6000.txt:2026-01-10T03:25:03.8675910Z 14.93s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-64___py3.13__13.1.0__local__H100.txt:2026-01-10T03:10:38.9082711Z 13.25s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-64___py3.13__13.1.0__local__rtxpro6000.txt:2026-01-10T03:24:43.8860109Z 13.10s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-64___py3.14__13.0.2__local__l4.txt:2026-01-10T03:09:43.3080030Z 12.62s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-64___py3.14__13.1.0__local__l4.txt:2026-01-10T03:13:13.1600312Z 13.46s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-64___py3.14t__13.0.2__local__l4.txt:2026-01-10T03:09:43.4362977Z 12.80s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-64___py3.14t__13.1.0__local__l4.txt:2026-01-10T03:13:38.5521286Z 14.57s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-aarch64___py3.10__12.9.1__local__a100.txt:2026-01-10T03:09:59.0165735Z 11.79s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-aarch64___py3.10__13.0.2__wheels__a100.txt:2026-01-10T03:20:22.5541190Z 12.06s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-aarch64___py3.10__13.1.0__wheels__a100.txt:2026-01-10T03:21:11.9424489Z 12.05s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-aarch64___py3.11__12.9.1__wheels__a100.txt:2026-01-10T03:15:14.1405192Z 11.53s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-aarch64___py3.11__13.0.2__local__a100.txt:2026-01-10T03:11:00.9066452Z 11.50s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-aarch64___py3.11__13.1.0__local__a100.txt:2026-01-10T03:10:54.0936687Z 11.52s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-aarch64___py3.12__13.0.2__wheels__a100.txt:2026-01-10T03:24:02.4632216Z 11.75s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-aarch64___py3.12__13.1.0__wheels__a100.txt:2026-01-10T03:17:44.4896939Z 11.64s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-aarch64___py3.13__12.9.1__wheels__a100.txt:2026-01-10T03:08:55.4501852Z 11.70s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-aarch64___py3.13__13.0.2__local__a100.txt:2026-01-10T03:17:36.9852261Z 11.58s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-aarch64___py3.13__13.1.0__local__a100.txt:2026-01-10T03:15:28.9082362Z 11.78s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-aarch64___py3.14__13.0.2__local__a100.txt:2026-01-10T03:10:45.3140851Z 11.76s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-aarch64___py3.14__13.1.0__local__a100.txt:2026-01-10T03:14:46.4410166Z 11.69s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-aarch64___py3.14t__13.0.2__local__a100.txt:2026-01-10T03:10:52.6658849Z 12.05s call     tests/test_helpers.py::test_patterngen_seeds
Test_linux-aarch64___py3.14t__13.1.0__local__a100.txt:2026-01-10T03:13:01.4328526Z 12.07s call     tests/test_helpers.py::test_patterngen_seeds
Test_win-64___py3.10__12.9.1__wheels__rtx2080__WDDM_.txt:2026-01-10T03:26:48.1889136Z 506.60s call     tests/test_helpers.py::test_patterngen_seeds
Test_win-64___py3.10__13.0.2__local__rtxpro6000__TCC_.txt:2026-01-10T03:44:19.9462869Z 702.69s call     tests/test_helpers.py::test_patterngen_seeds
Test_win-64___py3.10__13.1.0__local__rtxpro6000__TCC_.txt:2026-01-10T03:44:51.8427769Z 685.93s call     tests/test_helpers.py::test_patterngen_seeds
Test_win-64___py3.11__12.9.1__local__v100__MCDM_.txt:2026-01-10T03:30:23.4031313Z 763.77s call     tests/test_helpers.py::test_patterngen_seeds
Test_win-64___py3.11__13.0.2__wheels__rtx4090__WDDM_.txt:2026-01-10T03:27:44.5518977Z 506.31s call     tests/test_helpers.py::test_patterngen_seeds
Test_win-64___py3.11__13.1.0__wheels__rtx4090__WDDM_.txt:2026-01-10T03:26:54.8081465Z 529.50s call     tests/test_helpers.py::test_patterngen_seeds
Test_win-64___py3.12__12.9.1__wheels__l4__MCDM_.txt:2026-01-10T03:30:36.6552595Z 886.23s call     tests/test_helpers.py::test_patterngen_seeds
Test_win-64___py3.12__13.0.2__local__a100__TCC_.txt:2026-01-10T03:26:41.4786805Z 522.65s call     tests/test_helpers.py::test_patterngen_seeds
Test_win-64___py3.12__13.1.0__local__a100__TCC_.txt:2026-01-10T03:27:27.1592087Z 521.73s call     tests/test_helpers.py::test_patterngen_seeds
Test_win-64___py3.13__12.9.1__local__l4__TCC_.txt:2026-01-10T03:29:59.8609283Z 730.19s call     tests/test_helpers.py::test_patterngen_seeds
Test_win-64___py3.13__13.0.2__wheels__rtxpro6000__MCDM_.txt:2026-01-10T03:47:03.2483243Z 714.83s call     tests/test_helpers.py::test_patterngen_seeds
Test_win-64___py3.13__13.1.0__wheels__rtxpro6000__MCDM_.txt:2026-01-10T03:40:54.4933333Z 743.69s call     tests/test_helpers.py::test_patterngen_seeds
Test_win-64___py3.14__12.9.1__wheels__v100__TCC_.txt:2026-01-10T03:24:08.3370060Z 577.15s call     tests/test_helpers.py::test_patterngen_seeds
Test_win-64___py3.14__13.0.2__local__l4__MCDM_.txt:2026-01-10T03:35:16.9448794Z 930.24s call     tests/test_helpers.py::test_patterngen_seeds
Test_win-64___py3.14__13.1.0__local__l4__MCDM_.txt:2026-01-10T03:34:33.8842361Z 945.29s call     tests/test_helpers.py::test_patterngen_seeds
Test_win-64___py3.14t__12.9.1__local__l4__TCC_.txt:2026-01-10T03:30:01.8941727Z 767.32s call     tests/test_helpers.py::test_patterngen_seeds
Test_win-64___py3.14t__13.0.2__wheels__a100__MCDM_.txt:2026-01-10T03:29:51.7362946Z 623.74s call     tests/test_helpers.py::test_patterngen_seeds
Test_win-64___py3.14t__13.1.0__wheels__a100__MCDM_.txt:2026-01-10T03:27:48.3543004Z 757.88s call     tests/test_helpers.py::test_patterngen_seeds
```

### How to Reproduce

Download CI logs.

### Expected behavior

win-64 runtimes comparable to linux-64

### Operating System

_No response_

### nvidia-smi output

_No response_

### PR DESCRIPTION
**Pull request by Andy-Jost**
## Summary
- Replace exhaustive O(n²) pairwise seed testing with prime-stride sampling
- Reduces iterations from ~32,640 to ~100-150 while maintaining meaningful coverage
- Removes the Windows skip marker added in #1456, re-enabling the test on Windows

Closes #1455

### CODE REVIEW

File: cuda_core/tests/test_helpers.py [hunk 1]
```diff
@@ -47,10 +47,6 @@ def test_latchkernel():
     log("done")
 
 
-@pytest.mark.skipif(
```
Comment (Andy-Jost) [inline cuda_core/tests/test_helpers.py:50]: I could leave this in if we want to continue skipping this test on Windows. It is not important to test this on every platform IMO.

**Comment by Andy-Jost** [comment]
The test runs in 0.19s on Linux.

File: cuda_core/tests/test_helpers.py (hunk 1 above)
Comment (rwgk) [reply cuda_core/tests/test_helpers.py:50]: I'm 100% for removing this.

File: cuda_core/tests/test_helpers.py [hunk 2]
```diff
@@ -62,11 +58,13 @@ def test_patterngen_seeds():
     buffer = make_scratch_buffer(device, 0, NBYTES)
 
     # All seeds are pairwise different.
+    # We test a sampling of values because exhaustive testing is too slow,
+    # especially on Windows. See https://github.com/NVIDIA/cuda-python/issues/1455
     pgen = PatternGen(device, NBYTES)
-    for i in range(256):
+    for i in range(0, 256, 17):
```
Comment (rwgk) [inline cuda_core/tests/test_helpers.py]: This isn't covering the usual trouble-maker corner case 1 anymore.

This would sample around the usual suspects (0, 1 corner case, then a couple around powers of 2):

```
(0, 1, 2, 3, 4, 5, 31, 32, 33, 127, 128, 129)
```

Comment (Andy-Jost) [reply cuda_core/tests/test_helpers.py]: OK, I included values <5 since those are the most common. I don't think there's anything special about powers of two in this case.
